cpp/include/DataStorm/Node.h

```diff
@@ -38,47 +38,115 @@ namespace DataStorm
         /**
          * Construct a DataStorm node.
          *
-         * A node is the main DataStorm object. It is required to construct topics. The constructor initializes
-         * the Ice communicator using the given arguments. If the communicator creation fails, an Ice exception is
-         * raised.
+         * A node is the main DataStorm object. It is required to construct topics. The node uses the given Ice
+         * communicator.
+         *
+         * @param argc The number of arguments in argv.
+         * @param argv The configuration arguments.
+         * @param configFile The path to an optional Ice configuration file.
+         * @param customExecutor An optional executor used to execute user callbacks, if no callback executor is
+         * provided the Node will use the default callback executor that executes callback in a dedicated thread.
          *
-         * @param iceArgs Arguments which are passed to the Ice::initialize function.
          */
-        template<class... T> Node(T&&... iceArgs) : _ownsCommunicator(true)
+        Node(
+            int& argc,
+            const char* argv[],
+            std::optional<std::string_view> configFile = std::nullopt,
+            std::function<void(std::function<void()> call)> customExecutor = nullptr);
+
+        /**
+         * Construct a DataStorm node.
+         *
+         * A node is the main DataStorm object. It is required to construct topics. The node uses the given Ice
```

Comment:
This constructor (like most Node constructors) does not accept a communicator. So:
> The node uses the given Ice communicator.

is not correct.

Reply:
Fixed.